api-rights-bridge/src/test/java/org/xwiki/contrib/rights/internal/DefaultRightsWriterTest.java

```diff
@@ -125,10 +126,10 @@ void setUp()
     }
 
     @Test
-    void saveWhenNoRule() throws XWikiException
+    void saveWhenRuleIsNull() throws XWikiException
     {
         DocumentReference documentReference = new DocumentReference("xwiki", "S", "P");
-        this.rightsWriter.saveRules(Collections.emptyList(), documentReference);
+        this.rightsWriter.saveRules(null, documentReference);
```

Comment:
My guess here is that the original intention of the changes done for RIGHTSAPI-29 may have been to prevent the rights storage document from being created and saved if it doesn't exist and there is no rule to be saved, which makes sense as an expectation (it's not really what was tested in this test, though).

However, the way it was implemented (and tested) is not really doing that, as RIGHTSAPI-39 explains.

What I propose here is to implement that, and test it accordingly. Note the rights storage document is different depending on whether the rights are set on a terminal or non-terminal page, so I guess we'll need tests for each case - actually, the test we want is only for the case of a non-terminal page, which is the only case when a new storage document for rights may be created.
This test above is correct, we'll just need new ones.

Reply:
Created a second test for non-terminal page, which should be ok. And also improved this test to check on the call `saveDocument` instead of `getDocument`

But for this test I'm a bit confused because I don't really understand what you expect for the terminal page test:
1. We change the implementation, so before saving we check if there are some rights rule in the document. In case of empty right we do nothing.
2. We can accept that in the case of terminal page, we save the document with no changes, in case of `saveRules` call with empty right list parameter on a document without right list. In this case we can keep the test as it is or just remove it.

Reply:
Not 100% sur about what you was expecting on this. I tried to push a improvement on this side.